lib/middleware/getLicenceSeatsInfo.js

```diff
@@ -13,23 +14,25 @@ module.exports = (req, res, next) => {
 	return Promise.all([
 		accessLicenceClient.getLicenceInfo(req.licenceId)
 			.then(currentLicence => {
-				logger.info({operation, subOp: 'getLicenceInfo', licenceId: req.licenceId, seatLimit: currentLicence.seatLimit});
-				req.licenceSeatsInfo.seatLimit = currentLicence.seatLimit || -1;
+				logger.info({operation, subOp: '😾 getLicenceInfo', licenceId: req.licenceId, seatLimit: currentLicence.seatLimit});
+				req.licenceSeatsInfo.seatLimit = (currentLicence.seatLimit) ? currentLicence.seatLimit : -1;
 			})
 			.catch(err => {
 				logger.error({operation, subOp: 'getLicenceInfo', licenceId: req.licenceId, err});
 				return err;
 			}),
 
-		accessLicenceClient.getSeats(req.licenceId)
-			.then(currentSeats => {
-				const seatCount = Array.isArray(currentSeats) ? currentSeats.length : 0;
-				logger.info({operation, subOp: 'getSeats', licenceId: req.licenceId, allocatedSeatCount: seatCount});
-				req.licenceSeatsInfo.seatsAllocated = seatCount || 0;
+			licenceDataClient.getFilteredUserList(req.licenceId, req.apiAuthToken)
+			.then(licenceSeats => {
+				logger.info({operation, subOp: 'getSeats', licenceId: req.licenceId});
+				req.licenceSeatsInfo.seatsAllocated = Array.isArray(licenceSeats.seatHolders) ? licenceSeats.seatHolders.length : 0;
```

Comment:
is that API call not paginated at all? I would've thought it would be

Reply:
You're right, thougth if defaulted to all if no parma was set but it's actually 50 ☹️

Reply:
So will it return a whole list of ~10,000 users if needs be?

Reply:
Will now, upto 50,000 users. Was looking at calling getFilteredUserList recusively but the seatLimit value may not exist if the licences is effectively unlimited so not sure how we'd calculate the number of requests required. Our largest known licence is currentlly 13K. I could continue working on a solution but chosen to add this to the Kat tech debt given the best option for the future would be to refactor our api request to use the ft-core graphQL.